scripts/setup.sh

```diff
@@ -0,0 +1,232 @@
+    echo "  #     # #    # #    # #      #    #   #     #     #   #   #      #      #   #  #    # ##  ## #   #   #    # #    # "
+    echo "   #####   ####   ####  #      #    #   #     #     #   #   #      ###### #    #  ####  #    # #   #    ####  #    # "
+    echo ""                                                                                                           
+    sleep 1
+    echo -e "${NC}"
+    echo -e "🚀 ${BLUE}One-Click Docker Setup${NC} 🚀"
+    echo
+}
+
+# Detect Docker Compose version
+detect_docker_compose() {
+    # Check Docker
+    if ! command -v docker &> /dev/null; then
+        echo_error "Docker is not installed. Please install Docker first."
+        echo_info "Visit https://docs.docker.com/get-docker/ for installation instructions."
+        exit 1
+    fi
+    echo_success "Docker is installed."
+
+    # Check Docker Compose
+    if docker compose version &> /dev/null; then
+        DOCKER_COMPOSE="docker compose"
+        echo_success "Docker Compose is installed."
+    else
+        echo_error "Docker Compose is not installed. Please install Docker Compose(Alternatively use Orbstack) first."
+        echo_info "Visit https://docs.docker.com/compose/install/ for installation instructions."
+        exit 1
+    fi
+}
+
+check_prerequisites() {
+    # Check curl
+    if ! command -v curl &> /dev/null; then
+        echo_error "curl is not installed. Please install curl to proceed."
+        exit 1
+    fi
+    echo_success "curl is installed."
+    
+    # Check ports
+    required_ports=(8080 9000 9050 5432 6379)
+    unavailable_ports=()
+    
+    for port in "${required_ports[@]}"; do
+        if command -v nc &> /dev/null; then
+            if nc -z localhost "$port" 2>/dev/null; then
+                unavailable_ports+=("$port")
+            fi
+        elif command -v lsof &> /dev/null; then
+            if lsof -i :"$port" &> /dev/null; then
+                unavailable_ports+=("$port")
+            fi
+        else
+            echo_warning "Neither nc nor lsof available to check ports. Skipping port check."
+            break
+        fi
+    done
+    
+    if [ ${#unavailable_ports[@]} -ne 0 ]; then
+        echo_warning "The following ports are already in use: ${unavailable_ports[*]}"
+        echo_warning "This might cause conflicts with Hyperswitch services."
+        read -p "Do you want to continue anyway? (y/n): " -n 1 -r
+        echo
+        if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+            exit 1
+        fi
+    else
+        echo_success "All required ports are available."
+    fi
+}
+
+setup_config() {
+    if [ ! -f "config/docker_compose.toml" ]; then
+        if [ -f "config/config.example.toml" ]; then
+            echo "Creating docker_compose.toml from example config..."
+            cp config/config.example.toml config/docker_compose.toml
+            echo_success "Configuration file created."
```

Comment:
Let's not try to create Docker Compose config file from example config, it might be a bad idea, because the host names for some of the services may differ, etc.

If we can find `config/docker_compose.toml` we proceed, otherwise, we can error out.